create.ps1

```diff
@@ -1,194 +1,254 @@
+        )
+
+        $SplatParams = @{
+            Uri    = "$($Config.BaseUrl)/Users"
+            Method = 'Post'
+            Body   = $Body
         }
-    } catch {
-        $ex = $PSItem
-        if ( $($ex.Exception.GetType().FullName -eq 'Microsoft.PowerShell.Commands.HttpResponseException') -or $($ex.Exception.GetType().FullName -eq 'System.Net.WebException')) {
-            $errorMessage = Resolve-HTTPError -Error $ex
-            $auditMessage = "Account for '$($personObj.DisplayName)' not created. Error: $errorMessage"
-        } else {
-            $auditMessage = "Account for '$($personObj.DisplayName)' not created. Error: $($ex.Exception.Message)"
+
+        if (-Not ($ActionContext.DryRun -eq $True)) {
+            $UserResponse = Invoke-RestMethod @LisaRequest @SplatParams
+
+            $OutputContext.AccountReference = $UserResponse.objectId
+            $Account | Add-Member -NotePropertyName 'password' -NotePropertyValue $UserResponse.temporaryPassword
+        }
+        else {
+            Write-Verbose -Verbose ($Body | ConvertTo-Json)
+
+            $Account | Add-Member -NotePropertyName 'password' -NotePropertyValue "FakePassword"
         }
+
+        #Update the user with all other props
+        $Body = $Account | Select-Object -Property *, 'accountEnabled' -ExcludeProperty @(
+            'changePasswordNextSignIn', 'usageLocation', 'preferredLanguage',
+            'givenName', 'surName', 'displayName', 'userPrincipalName',
+            'password'
+        )
+
+        # Force the account disabled
+        $Body.accountEnabled = $False
+
+        $SplatParams = @{
+            Uri    = "$($config.BaseUrl)/Users/$($OutputContext.AccountReference)/bulk"
+            Method = 'Patch'
+            Body   = $Body
+        }
+
+        if (-Not ($ActionContext.DryRun -eq $True)) {
+            [void] (Invoke-RestMethod @LisaRequest @SplatParams)
+        }
+        else {
+            Write-Verbose -Verbose ($Body | ConvertTo-Json)
+        }
+
+        # Set the manager
+        if ($PersonContext.References.ManagerAccount) {
+            $SplatParams = @{
+                Uri    = "$($Config.BaseUrl)/Users/$($OutputContext.AccountReference)/Manager"
+                Method = 'Put'
+                Body   = $PersonContext.References.ManagerAccount
+            }
+
+            if (-Not ($ActionContext.DryRun -eq $True)) {
+                [void] (Invoke-RestMethod @LisaRequest @SplatParams)
+            }
+
+            Write-Verbose -Verbose "Added Manager $($Manager.displayName) to '$($Person.DisplayName)'"
+        }
+
+        $AuditLogs.Add([PSCustomObject]@{
+                Action  = "CreateAccount" # Optionally specify a different action for this audit log
+                Message = "Created account for '$($Person.DisplayName)'. Id: $($OutputContext.AccountReference)"
+                IsError = $False
+            })
+
+        $OutputContext.Success = $True
```

Comment:
HelloID provides the `$OutputContext.Success` as false by default. This should be false unless the user is created and provisioned. If the script is successful by design and no auditlogs are provided because the script failed, this will tell helloid the script was succesful. A big no no for me.

this seems like a serious risk to take. (would save performance for not having to retry the action i guess)

Reply:
agreed, then i would propose something like this:
```powershell
$outputContext.Success = $false
try {
    # do logic
    # auditlog iserror = $false
}
catch {
    # upon error
    # auditlog iserror = $true
}
finally {
    # Check if auditLogs contains errors, if no errors are found, set success to true
    if (!($outputContext.AuditLogs.IsError -contains $true)) {
        $outputContext.Success = $true
    } 
}
```